### PR TITLE
Adding fileSet instead of TestClass

### PR DESCRIPTION
**Pull request by litzenberger**
screen shot 2018-12-05 at 6 38 58 am

### CODE REVIEW

File: tests/lib/fixtures.js [hunk 1]
```diff
@@ -1,8 +1,37 @@
...
+  8 * MB,
+  64 * MB,
+  128 * MB,
+  512 * MB,
+  GB,
+  10 * GB,
+  100 * GB,
+  1000 * GB
+]
+*/
```
Comment (mcollina) [inline tests/lib/fixtures.js]: why are these commented out?

File: tests/lib/fixtures.js [hunk 2]
```diff
@@ -1,8 +1,37 @@
...
 
 module.exports = {
   smallFile: path.join(__dirname, '../fixtures/200Bytes.txt'),
-  largeFile: path.join(__dirname, '../fixtures/1.2MiB.txt')
+  largeFile: path.join(__dirname, '../fixtures/1.2MiB.txt'),
+  OneKBFile: crypto.randomBytes(KB),
+  OneMBFile: crypto.randomBytes(MB),
```
Comment (mcollina) [inline tests/lib/fixtures.js]: I think we should add a 4 MB and 64MB  at least for now.

File: tests/lib/fixtures.js [hunk 3]
```diff
@@ -1,8 +1,37 @@
...
+  OneGBFile: crypto.randomBytes(GB)
```
Comment (mcollina) [inline tests/lib/fixtures.js]: I would prefer if those file are written down to disk on first use, and then referenced with paths.
Holding a file of 1GB in memory might be too big.

Comment (litzenberger) [reply tests/lib/fixtures.js]: yep, we are going to pre-generate them

File: tests/lib/fixtures.js (hunk 2 above)
Comment (litzenberger) [reply tests/lib/fixtures.js]: yes,  here is the list we are using.

```
const files = [
  { size: KB, name: 'KB' },
  { size: 62 * KB, name: '62KB' },
  { size: 64 * KB, name: '64KB' },
  { size: 512 * KB, name: '512KB' },
  { size: 768 * KB, name: '768KB' },
  { size: 1023 * KB, name: '1023KB' },
  { size: 1023 * KB, name: 'MB' },
  { size: 4 * MB, name: '4MB' },
  { size: 8 * MB, name: '8MB' },
  { size: 64 * MB, name: '64MB' },
  { size: 128 * MB, name: '128MB' },
  { size: 512 * MB, name: '512MB' },
  { size: MB, name: 'GB' },
  { size: 10 * GB, name: '10GB' },
  { size: 100 * GB, name: '100GB' },
  { size: 1000 * GB, name: '1000GB' }
]
```

File: tests/lib/fixtures.js [hunk 4]
```diff
@@ -1,8 +1,9 @@
...
 
 const path = require('path')
-
 module.exports = {
-  smallFile: path.join(__dirname, '../fixtures/200Bytes.txt'),
-  largeFile: path.join(__dirname, '../fixtures/1.2MiB.txt')
+  OneKBFile: path.join(__dirname, '../fixtures/KB.txt'),
+  OneMBFile: path.join(__dirname, '../fixtures/MB.txt'),
+  One4MBFile: path.join(__dirname, '../fixtures/4MB.txt'),
+  One64MBFile: path.join(__dirname, '../fixtures/64MB.txt')
```
Comment (mcollina) [inline tests/lib/fixtures.js]: Files are defined here.

File: tests/util/create-files.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+]
+
+for (let file of files) {
+  write(crypto.randomBytes(file.size), file.name)
+}
+
+async function write (data, name) {
+  await fsWriteFile(path.join(__dirname, `../fixtures/${name}.txt`), data)
+  console.log(`File ${name} created.`)
+}
```
Comment (mcollina) [inline tests/util/create-files.js]: Files are created here. I think we should have only one source of truth where file sizes are defined, and then imported in both places.

Comment (litzenberger) [reply tests/util/create-files.js]: yep, will fix that.

**Pull request merged**